File: solutions/java/src/test/java/playground/algorithm/LongestCommonPrefixTest.java
package playground.algorithm;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class LongestCommonPrefixTest {

    @Test
    void case_1() {
        LongestCommonPrefix solution = new LongestCommonPrefix();

        String prefix = solution.longestCommonPrefix(new String[]{"flower", "flow", "flight"});

        assertEquals("fl", prefix);
    }

    @Test
    void case_2() {
        LongestCommonPrefix solution = new LongestCommonPrefix();

        String prefix = solution.longestCommonPrefix(new String[]{"dog", "racecar", "car"});

        assertEquals("", prefix);
    }

    @Test
    void case_3() {
        LongestCommonPrefix solution = new LongestCommonPrefix();

        String prefix = solution.longestCommonPrefix(new String[]{});

        assertEquals("", prefix);
    }
}

File: solutions/go/src/playground/algorithm/longest_common_prefix_test.go
package algorithm

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestLongestCommonPrefixCase1(t *testing.T) {
	assert.Equal(t, "fl", longestCommonPrefix([]string{"flower", "flow", "flight"}))
}

func TestLongestCommonPrefixCase2(t *testing.T) {
	assert.Equal(t, "", longestCommonPrefix([]string{"dog", "racecar", "car"}))
}

func TestLongestCommonPrefixCase3(t *testing.T) {
	assert.Equal(t, "a", longestCommonPrefix([]string{"a"}))
}

func TestLongestCommonPrefixCase4(t *testing.T) {
	assert.Equal(t, "", longestCommonPrefix([]string{}))
}
